source/index-management/tests/README.md

```diff
@@ -231,3 +231,108 @@ This test fails if it times out waiting for the deletion to succeed.
+    condition is satisfied and store the value in a variable `index1`:
+
+    - An index with the `name` of `test-search-index-case7-implicit` is present and the index has a field `queryable`
+      with a value of `true`.
+
+05. Assert that `index1` has a property `type` whose value is `search`.
+
+06. Create a new search index on `coll0` with the `createSearchIndex` helper. Use the following definition:
+
+    ```typescript
+
+      {
+        name: 'test-search-index-case7-explicit',
+        type: 'search',
+        definition: {
+          mappings: { dynamic: false }
+        }
+      }
+    ```
+
+07. Assert that the command returns the name of the index: `"test-search-index-case7-explicit"`.
+
+08. Run `coll0.listSearchIndexes('test-search-index-case7-explicit')` repeatedly every 5 seconds until the following
+    condition is satisfied and store the value in a variable `index2`:
+
+    - An index with the `name` of `test-search-index-case7-explicit` is present and the index has a field `queryable`
+      with a value of `true`.
+
+09. Assert that `index2` has a property `type` whose value is `search`.
+
+10. Create a new vector search index on `coll0` with the `createSearchIndex` helper. Use the following definition:
+
+    ```typescript
+
+      {
+        name: 'test-search-index-case7-vector',
+        type: 'vectorSearch',
+        definition: {
+          fields: [
+             {
+                 type: 'vector',
+                 path: 'plot_embedding',
+                 numDimensions: 1536,
+                 similarity: 'euclidean',
+             },
+          ]
+        }
+      }
+    ```
+
+11. Assert that the command returns the name of the index: `"test-search-index-case7-vector"`.
+
+12. Run `coll0.listSearchIndexes('test-search-index-case7-vector')` repeatedly every 5 seconds until the following
+    condition is satisfied and store the value in a variable `index3`:
+
+    - An index with the `name` of `test-search-index-case7-vector` is present and the index has a field `queryable` with
+      a value of `true`.
+
+13. Assert that `index3` has a property `type` whose value is `vectorSearch`.
+
+#### Case 8: Driver requires explicit type to create a vector search index
+
+1. Create a collection with the "create" command using a randomly generated name (referred to as `coll0`).
+
+2. Create a new vector search index on `coll0` with the `createSearchIndex` helper. Use the following definition:
+
+   ```typescript
+
+     {
+       name: 'test-search-index-case8-error',
+       definition: {
+         fields: [
+            {
+                type: 'vector',
+                path: 'plot_embedding',
+                numDimensions: 1536,
+                similarity: 'euclidean',
+            },
+         ]
+       }
+     }
+   ```
+
+3. Assert that the command throws an exception containing the string "Attribute mappings missing" due to the `mappings`
```

Comment:
Does this test actually confirm that `type` is required?  The error `Attribute mappings missing` seems to indicate that `mappings` must be present.  I added a `mappings` and confirmed this with a patch: https://spruce.mongodb.com/task/mongo_node_driver_next_rhel8_test_search_indexes_test_search_index_helpers_patch_3ae4ff5d3ebb3ebb251eea650e81b8dd4bf444b9_661d5383b25584000708ce66_24_04_15_16_19_48/logs?execution=0

Reply:
It confirms that `type` is required to create a vector search index. Vector search indexes use `fields` instead of `mappings`, but the error returned by the server doesn't provide this information.